test/hierarchy_test.rb

```diff
@@ -216,6 +216,16 @@ def test_finds_leaves
     assert_equal [child2, child3, grandchild1], root.descendents.leaves.order(:created_at).to_a
   end
 
+  def test_finds_all_leaves
+    root = TreeNode.create!
+    child1 = TreeNode.create!(parent: root)
+    child2 = TreeNode.create!(parent: root)
+    child3 = TreeNode.create!(parent: root)
+    grandchild1 = TreeNode.create!(parent: child1)
+
+    assert_equal [child2, child3, grandchild1], TreeNode.leaves.order(:created_at).to_a
```

Comment:
What would happen if you create another `root`? The `parent` would be `null` but the it would also be a `leaf` right? `root` and `leaf` at the same time.